src/BrazeKit-dev.js

```diff
@@ -290,6 +335,157 @@ var constructor = function () {
+    // a purchase event or a non-purchase commerce event
+    function logCommerceEvent(event) {
+        if (
+            event.EventCategory === mParticle.CommerceEventType.ProductPurchase
+        ) {
+            reportEvent = logPurchaseEvent(event);
+            return reportEvent === true;
+        } else {
+            reportEvent = logNonPurchaseCommerceEvent(event);
+            return reportEvent === true;
+        }
+    }
+
+    // A non-purchase commerce event can either log a single event with all products
+    // or one event per product when the commerce event is expanded
+    function logNonPurchaseCommerceEvent(event) {
+        if (bundleCommerceEventData) {
+            return logNonPurchaseCommerceEventWithProducts(event);
+        } else {
+            return logExpandedNonPurchaseCommerceEvents(event);
+        }
+    }
+
+    function logNonPurchaseCommerceEventWithProducts(mpEvent) {
+        const commerceEventAttrs = {};
+
+        try {
+            switch (mpEvent.EventCategory) {
+                case CommerceEventTypes.PromotionClick:
+                case CommerceEventTypes.PromotionView:
+                    commerceEventAttrs.promotions = addPromotions(
+                        mpEvent.PromotionAction
+                    );
+                    break;
+                case CommerceEventTypes.ProductImpression:
+                    commerceEventAttrs.impressions = addImpressions(
+                        mpEvent.ProductImpressions
+                    );
+                    break;
+                default:
+                    if (mpEvent.ProductAction.ProductList) {
+                        commerceEventAttrs.products = addProducts(
+                            mpEvent.ProductAction.ProductList
+                        );
+                    }
+                    var transactionId = mpEvent.ProductAction.TransactionId;
+                    if (transactionId) {
+                        commerceEventAttrs['Transaction Id'] = transactionId;
+                    }
+            }
+
+            var sanitizedProperties = getSanitizedCustomProperties(
+                mpEvent.EventAttributes
+            );
+
+            const brazeEvent = {
+                EventName: mpEvent.EventName,
+                EventAttributes: mergeObjects(
+                    commerceEventAttrs,
+                    sanitizedProperties
+                ),
+            };
+
+            reportEvent = logBrazeEvent(brazeEvent);
+            return reportEvent;
+        } catch (err) {
+            return 'Error logging commerce event' + err.message;
+        }
+    }
+
+    function addPromotions(promotionAction) {
+        if (promotionAction && promotionAction.PromotionList) {
+            return promotionAction.PromotionList;
+        }
+        return [];
+    }
+
+    function addImpressions(productImpressions) {
+        if (productImpressions.length) {
```

Comment:
I see we're checking to see if a property is an empty array in different ways. Can we just leverage the `isEmpty` utility from core to make things cleaner?